haas/api.py

```diff
@@ -529,56 +565,111 @@ def network_delete(network):
     db = model.Session()
     network = _must_find(db, model.Network, network)
 
-    if network.nics:
+    if len(network.attachments) != 0:
         raise BlockedError("Network still connected to nodes")
     if network.hnics:
         raise BlockedError("Network still connected to headnodes")
-    if network.scheduled_nics:
-        raise BlockedError("Network scheduled to become connected to nodes.")
+    if len(network.scheduled_nics) != 0:
+        raise BlockedError("There are pending actions on this network")
     if network.allocated:
-        driver_name = cfg.get('general', 'driver')
-        driver = importlib.import_module('haas.drivers.' + driver_name)
-        driver.free_network_id(db, network.network_id)
+        get_network_allocator().free_network_id(db, network.network_id)
 
     db.delete(network)
     db.commit()
 
 
-                            # Port code #
-                            #############
+@rest_call('GET', '/network/<network>')
+def show_network(network):
+    """Show details of a network.
+
+    Returns a JSON object representing a network.
+    The object will have at least the following fields:
+        * "name", the name/label of the network (string).
+        * "creator", the name of the project which created the network, or
+          "admin", if it was created by an administrator.
+    It may also have the fields:
+        * "access" -- if this is present, it is the name of the project which
+          has access to the network. Otherwise, the network is public.
+    """
+    db = model.Session()
+    allocator = get_network_allocator()
 
+    network = _must_find(db, model.Network, network)
+    return json.dumps({
+        'name': network.label,
+        'channels': allocator.legal_channels_for(db, network.network_id),
+    })
 
-@rest_call('PUT', '/port/<path:port>')
-def port_register(port):
+
+@rest_call('PUT', '/switch/<switch>', schema=Schema({
+    'type': basestring,
+    Optional(object): object,
+}))
+def switch_register(switch, type, **kwargs):
+    db = model.Session()
+    _assert_absent(db, model.Switch, switch)
+
+    cls = concrete_class_for(model.Switch, type)
+    if cls is None:
+        raise BadArgumentError('%r is not a valid switch type.' % type)
+    cls.validate(kwargs)
+    obj = cls(**kwargs)
+    obj.label = switch
+    obj.type = type
+
+    db.add(obj)
+    db.commit()
```

Comment:
It's a little unfortunate that we're shadowing the Python built-in 'type' in this function, but I don't see any good way of avoiding that